tools/scoring/score/dimensions/dimension.py

```diff
@@ -176,8 +180,123 @@ def is_entity_virtual(*, entity: EntityInstance) -> bool:
     """
     return entity.links is not None
 
+  @staticmethod
+  def match_virtual_entities(
+      *, solution_points_virtual: List[Tuple[Set[str], EntityType]],
+      proposed_points_virtual: List[Tuple[Set[str], EntityType]],
+      sort_candidates_by_key: str) -> Dict[float, List[Dict]]:
```

Comment:
it this a score?
what is inside the List[Dict] ?

Reply:
`float` is a subscore, i.e. how closely the matched entities correlated. The dicts are parameters by which those floats were calculated. (It's described in the docstring but I'll work on annotation.)